src/snowcli/cli/nativeapp/teardown_processor.py

```diff
@@ -0,0 +1,233 @@
+            show_obj_cursor = self._execute_query(
+                show_obj_query, cursor_class=DictCursor
+            )
+
+            if show_obj_cursor.rowcount is None:
+                raise SnowflakeSQLExecutionError(show_obj_query)
+
+            show_obj_row = find_row(
+                show_obj_cursor,
+                lambda row: row[NAME_COL] == unquote_identifier(self.package_name),
+            )
+
+            # If existing application is not found, exit gracefully
+            if show_obj_row is None:
+                log.info(
+                    f"Role {self.package_role} does not own any application packages with the name {self.package_name}!"
+                )
+                return
+
+            # 2. Check for the right owner
+            actual_owner = show_obj_row[OWNER_COL]
+            if actual_owner != unquote_identifier(self.package_role):
+                raise UnexpectedOwnerError(
+                    self.package_name, self.package_role, actual_owner
+                )
+
+            # 3. Check for versions in the application package
+            show_versions_query = (
+                f"show versions in application package {self.package_name}"
+            )
+            show_versions_cursor = self._execute_query(
+                show_versions_query, cursor_class=DictCursor
+            )
+            if show_versions_cursor.rowcount is None:
+                raise SnowflakeSQLExecutionError(show_versions_query)
+            if show_versions_cursor.rowcount > 0:
+                raise CouldNotDropApplicationPackageWithVersions()
+
+            # 4. Check distribution of the existing app pkg
+            actual_distribution = self.get_app_pkg_distribution_in_snowflake
+            if self.is_app_pkg_distribution_mismatched():
+                log.warning(
+                    dedent(
+                        f"""\
+                        Continuing to execute `snow app teardown` on app pkg {self.package_name} with distribution {actual_distribution}.
+                    """
+                    )
+                )
+
+            # 5. If distribution is internal, check for special comment
+            row_comment = show_obj_row[COMMENT_COL]
+            if actual_distribution == INTERNAL_DISTRIBUTION:
+                if row_comment == SPECIAL_COMMENT:
+                    self.needs_confirm = False
+                else:
+                    if self.needs_confirm and not auto_yes:
+                        log.warn(
+                            f"Application package {self.package_name} was not created by SnowCLI!"
+                        )
+            else:
+                if self.needs_confirm and not auto_yes:
+                    log.warn(
+                        f"Application package {self.package_name} in your Snowflake account has distribution property '{EXTERNAL_DISTRIBUTION}'"
+                    )
```

Comment:
I don't think this else clause is necessary. The log below will print out the package distribution anyway.

Reply:
True, but the log below does not tell what the problem/mismatch is despite all the info. It could be anything - owner, created_on etc. Keeping this log gives the user more information in a clear manner.